jwt/__init__.py

```diff
@@ -38,9 +38,24 @@ class EncodeError(Exception): pass
     'RS256': lambda msg, key, sig: key.verify(hashlib.sha256(msg).digest(), sig, 'sha256'),
     'RS384': lambda msg, key, sig: key.verify(hashlib.sha384(msg).digest(), sig, 'sha384'),
     'RS512': lambda msg, key, sig: key.verify(hashlib.sha512(msg).digest(), sig, 'sha512'),
-    'none': lambda msg, key, sig: True
+    'none': lambda msg, key, sig: check_none_verify(msg, key, sig)
 }
 
+SUPPORTED_ALGOS = tuple(verifying_methods.keys())
+ALLOWED_ALGOS = sorted(SUPPORTED_ALGOS)
+# Remove the 'none' type from our particular usage
+del ALLOWED_ALGOS[len(ALLOWED_ALGOS) - 1] 
+ALLOWED_ALGOS = tuple(ALLOWED_ALGOS)
+
+def check_none_verify(msg, key, sig):
+    """
+    If there is a signature included with a alg=none header then the
+    verification should fail
+    """
+    if sig:
+        return False
```

Comment:
what is this meant to protect against? Anyone can just remove the signature. By default, none-type JWTs have a missing signature.